### DISCUSSION TITLE
[ZK Stack] - Underpriced transactions to execute batches on Sepolia

### DISCUSSION THREAD

**Discussion by Archethect** [zkSync-Community-Hub/zksync-developers#407] [ZK Stack] [answered]
### Module Affected

- [ ] System contracts (era-system-contracts)
- [ ] Smart contracts (era-contracts)
- [ ] zkEVM (era-zk_evm)
- [X] Server (zksync-era)
- [ ] Prover (era-boojum)
- [ ] Fee withdrawer (era-fee-withdrawer)
- [ ] Block Explorer (clock-explorer)
- [ ] Portal - Wallet + Bridge (dapp-portal)
- [ ] Other (please specify below)

### Rust Version

1.75.0

### Operating System

- [X] Ubuntu
- [ ] MacOS
- [ ] Windows
- [ ] Other (Please specify below)

### Issue Description

While running ZK Stack connected to sepolia without prover we have long delays for the transactions that execute the batches on L1.

### Expected Behavior

We would expect to have proper gas estimation and retries for batch executions on L1 such that they get executed within minutes rather than hours.

### Current Behavior

Batch executes keep hanging. Occasionally the server tries to send a replacement transaction, but these seem to fail as well:

`[2mzksync_core::eth_sender::eth_tx_manager[0m[2m:[0m Error when sending new signed tx for tx 32, base_fee_per_gas 42297646025, priority_fee_per_gas: 2488320005: Ethereum gateway Error Request to ethereum gateway failed: RPC error: Error { code: ServerError(-32000), message: "replacement transaction underpriced", data: None }`

Is there an option to modify this behavior and make sure the gasprice can be set to x2 the current gasprice when it failed several times?

### Prior Research

- [X] Yes, I searched for similar issues and did not find any matching ones.

**Comment by zk-Lumi** [comment, marked as answer]
Hi @Archethect ,

There may have been a spike of gas price on Sepolia, and the system is designed to attempt to save fees. Should solve it's self. Mind confirming?

To force you can also find the eth_txs_history necessary tx and change DB value `base_fee_per_gas` to the desired one and it will be used next time (slightly increased).

**Reply by Archethect** [reply]
@zk-Lumi ,Yes I can confirm it auto resolved itself. So all good.
Thanks for the tip to force it through as well!
